src/transport.rs

```diff
@@ -180,6 +180,11 @@ impl Ord for Endpoint {
 
 #[derive(Debug, Clone, RustcDecodable, RustcEncodable)]
 pub enum Message {
+    /// After a connection is established, peers should exchange a handshake.
+    ///
+    /// Only after the handshake is exchanged, crust should generate new
+    /// connection events.
+    Handshake,
```

Comment:
I think `Handshake` should not be part of the `Message` enum as we'll never send `UserBlob` nor `Contacts` while handshaking and we'll never send `Handshake` while receiving `UserBlob` or `Contacts`.